File: 0200-number-of-islands/0200-number-of-islands.cpp
class Solution {
public:
    int numIslands(vector<vector<char>>& grid) {
        int num = 0;
        for(int i = 0; i < grid.size(); i++) {
            for(int k = 0; k < grid[0].size(); k++) {
                if(grid[i][k] == '1') {
                    recursive(grid, i, k);
                    num++;
                }
               
            }
        }
        return num;
    }
    void recursive(vector<vector<char>>& grid, int row, int col) {
        grid[row][col] = '0';
        
        if(row + 1 < grid.size() && grid[row + 1][col] == '1') {
            recursive(grid, row + 1, col);
        }
        if(row - 1 >= 0 && grid[row-1][col] == '1') {
            recursive(grid, row-1,col);
        }
        if(col + 1 < grid[0].size() && grid[row][col+1] == '1') {
            recursive(grid, row, col + 1);
        }
        if(col - 1 >= 0 && grid[row][col-1] == '1') {
            recursive(grid, row, col - 1);
        }
        
    }
};

File: 0200-number-of-islands/0200-number-of-islands.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        
        def dfs(r, c):

            if r < 0 or r >= len(grid) or c < 0 or c >= len(grid[0]):
                return
            if grid[r][c] == '0':
                return
            grid[r][c] = '0'
            dfs(r+1, c)
            dfs(r-1, c)
            dfs(r, c-1)
            dfs(r, c+1)

        total = 0
        for row in range(len(grid)):
            for col in range(len(grid[0])):
                if grid[row][col] == '1':
                    print('found')
                    total += 1 
                    dfs(row, col)
        return total
                
        